src/Action/RefundAction.php

```diff
@@ -50,17 +50,22 @@ public function execute($request): void
 
         $details = ArrayObject::ensureArrayObject($request->getModel());
 
-        if ($details['created_in_mollie']) {
-            $this->loggerAction->addLog('Received refund created in Mollie dashboard');
+        try {
+            $molliePayment = $this->mollieApiClient->payments->get($details['payment_mollie_id']);
+        } catch (ApiException $e) {
+            $this->loggerAction->addNegativeLog(sprintf('API call failed: %s', htmlspecialchars($e->getMessage())));
 
+            throw new \Exception(sprintf('API call failed: %s', htmlspecialchars($e->getMessage())));
+        }
+
+        if ($molliePayment->hasRefunds()) {
```

Comment:
Just for confirmation: we do not care about the refunds state, right?

Reply:
Yes, we do not care.